### PR TITLE
Add multiplayer

### PR DESCRIPTION
**Pull request by Resocram**

### CODE REVIEW

File: components/Pages/RoomIdPage.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import Cookies from 'js-cookie';
+
+const backendApiUrl = 'http://localhost:5000';
+
+class RoomIdPage extends React.Component {
```
Comment (stellaw1) [inline components/Pages/RoomIdPage.js]: can consider swapping to function component syntax `export default function RoomIdPage(props) {`
and setting state becomes like `const [roomId, setRoomId] = React.useState(props.roomId);` etc

File: components/Pages/RoomPage.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+const backendApiUrl = 'http://localhost:5000';
+class RoomPage extends React.Component {
```
Comment (stellaw1) [inline components/Pages/RoomPage.js:3]: same as above

File: server/server.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+      if (client.readyState === WebSocket.OPEN) {
+        client.send(JSON.stringify({ type: 'update_players', players: players }));
+      }
+    })
+  }
+
+  function getAllPlayers() {
+    const players = []
+    clients.forEach((value, client) => {
+      players.push(value.get('username'))
```
Comment (stellaw1) [inline server/server.js]: consider using a set `Array.from(new Set(players));` before returning b/c multiple tabs on the same browser leads to same username showing multiple time 

image

**Review by stellaw1 — APPROVED** [review]
LGTM🚀🚀🚀🚀🚀

File: server/server.js (hunk 3 above)
Comment (Resocram) [reply server/server.js]: :O this is actually a bigger problem since I allow for multiple usernames, I will add a check for unique username

**Pull request merged**